### PR TITLE
workspace-support

### PR DESCRIPTION
**Pull request by gautamprikshit1**
Made an attempt at #564 please check

### CODE REVIEW

File: cargo-shuttle/src/init.rs [hunk 1]
```diff
@@ -669,7 +670,12 @@ pub fn cargo_shuttle_init(path: PathBuf, framework: Framework) -> Result<()> {
     let mut dependencies = Table::new();
 
     // Set "shuttle-service" version to `[dependencies]` table
-    let manifest_path = find(Some(path.as_path())).unwrap();
+    let config = cargo::util::config::Config::default().unwrap();
```
Comment (chesedo) [inline cargo-shuttle/src/init.rs:673]: Nice!

I'm just not sure if `cargo_shuttle_init` is the correct function to update though. I think this function needs to be updated or removed (which is better)

https://github.com/shuttle-hq/shuttle/blob/79ff57e1a2e712748803d20e7aa51b1f9dbf72af/cargo-shuttle/src/lib.rs#L210-L214

**Review by chesedo — COMMENTED** [review]
Wow, that was easy. Thanks @gautamprikshit1 

Just confirm this is updating the correct function. The easiest would be to create a workspace with two crates:
1. A crate with a library function like `get_name()`
2. A crate that uses shuttle and crate 1 to get a name

Then from inside crate 1 run `cargo shuttle run`. If it fails then the wrong function might have been updated here

File: cargo-shuttle/src/init.rs (hunk 1 above)
Comment (gautamprikshit1) [reply cargo-shuttle/src/init.rs:673]: Have you run it already or shall I do it??

Comment (gautamprikshit1) [reply cargo-shuttle/src/init.rs:673]: https://docs.rs/project-root/latest/src/project_root/lib.rs.html#19-34

I think replacing existing function with this one will do.

Comment (chesedo) [reply cargo-shuttle/src/init.rs:673]: I'll leave the testing to you :smile:

**Pull request closed without merging**